jiant/scripts/download_data/datasets/xtreme.py

```diff
@@ -0,0 +1,372 @@
+                q_id = question_id.split("-")[-1]
+                for answer in qa["answers"]:
+                    a_start, a_text = answer["answer_start"], answer["text"]
+                    a_end = a_start + len(a_text)
+                    assert context[a_start:a_end] == a_text
+                lang2data[example_lang].append(
+                    {
+                        "paragraphs": [
+                            {
+                                "context": context,
+                                "qas": [
+                                    {"answers": qa["answers"], "question": question, "id": q_id}
+                                ],
+                            }
+                        ]
+                    }
+                )
+
+    for full_lang, lang in languages_dict.items():
+        task_name = f"tydiqa_{lang}"
+        task_data_path = py_io.create_dir(task_data_base_path, task_name)
+        train_path = os.path.join(task_data_path, f"tydiqa.{lang}.train.json")
+        py_io.write_json(
+            data={"data": data, "version": version}, path=train_path,
+        )
+        val_path = os.path.join(task_data_path, f"tydiqa.{lang}.dev.json")
+        os.rename(
+            src=os.path.join(
+                tydiqa_temp_path, "tydiqa-goldp-v1.1-dev", f"tydiqa-goldp-dev-{full_lang}.json"
+            ),
+            dst=val_path,
+        )
+        py_io.write_json(
+            data={
+                "task": "tydiqa",
+                "paths": {"train": train_path, "val": val_path},
+                "kwargs": {"language": lang},
+                "name": task_name,
+            },
+            path=os.path.join(task_config_base_path, f"{task_name}_config.json"),
+        )
+    shutil.rmtree(tydiqa_temp_path)
+
+
+def download_bucc2018_data_and_write_config(task_data_base_path: str, task_config_base_path: str):
+    bucc2018_temp_path = py_io.create_dir(task_data_base_path, "bucc_temp")
+    languages = "de fr ru zh".split()
+    for lang in languages:
+        download_utils.download_and_untar(
+            f"https://comparable.limsi.fr/bucc2018/bucc2018-{lang}-en.training-gold.tar.bz2",
+            bucc2018_temp_path,
+        )
+        download_utils.download_and_untar(
+            f"https://comparable.limsi.fr/bucc2018/bucc2018-{lang}-en.sample-gold.tar.bz2",
+            bucc2018_temp_path,
+        )
+    for lang in languages:
+        task_name = f"bucc2018_{lang}"
+        task_data_path = py_io.create_dir(task_data_base_path, task_name)
+        val_eng_path = os.path.join(task_data_path, f"{lang}-en.dev.en")
+        val_other_path = os.path.join(task_data_path, f"{lang}-en.dev.{lang}")
+        val_labels_path = os.path.join(task_data_path, f"{lang}-en.dev.gold")
+        test_eng_path = os.path.join(task_data_path, f"{lang}-en.test.en")
+        test_other_path = os.path.join(task_data_path, f"{lang}-en.test.{lang}")
+        # sample -> dev
+        # training -> test (yup, it's weird)
```

Comment:
what lines do these comments refer to?

Reply:
Bucc2018 doesn't have actual dev/test sets. Instead, the "sample" set is used as dev, and the "training" set is used as the test set. This comment explains the remapping.